src/main/java/ru/skypro/homework/service/impl/CommentServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+import ru.skypro.homework.repository.CommentRepository;
+import ru.skypro.homework.repository.UserRepository;
+import ru.skypro.homework.service.CommentService;
+
+import javax.validation.constraints.NotNull;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+@Service
+
+public class CommentServiceImpl implements CommentService {
+
+    private final CommentRepository commentRepository;
+    private final CommentMapper commentMapper;
+    private final AdRepository adRepository;
+    private final UserRepository userRepository;
+
+    @Override
+    public CommentDto createComment(Integer id,
+                                    CreateOrUpdateCommentDto createCommentDto,
+                                    Authentication authentication) {
+        Timestamp localDateTime = Timestamp.valueOf(LocalDateTime.now().truncatedTo(ChronoUnit.MINUTES));
+        Comment comment = new Comment();
+        Ad ad = adRepository.findAdById(id);
+        if (ad == null) {
+            return null;
+        }
+        comment.setText(createCommentDto.getText());
+        comment.setCreatedAt(localDateTime);
+        comment.setAd(ad);
+        comment.setAuthor(userRepository.findByEmail(authentication.getName()).orElseThrow(UserNotFoundException::new));
+        commentRepository.save(comment);
+        return commentMapper.commentToCommentDto(comment);
+    }
+    @Override
+    public CommentsDto getComments(Integer adId) {
+        List<Comment> comment = commentRepository.findByAd_Id(adId);
+        if (comment == null) {
+            return null;
+        }
+        List<CommentDto> commentList = comment.stream()
+                .map(commentMapper::commentToCommentDto)
+                .collect(Collectors.toList());
+        return new CommentsDto(commentList.size(), commentList);
+    }
+
+/*    @Override
+    public CommentDto addComment(Integer adId, CreateOrUpdateCommentDto commentDto) {
+        Comment comment = new Comment();
+        Ad ad = adRepository.findById(adId).orElseThrow(() -> new AdNotFoundException());
+
+        comment.setAd(ad);
+
+        commentMapper.createOrUpdateDtoFromComment(comment);
+
+        Comment savedComment = commentRepository.save(comment);
+
+        return commentMapper.commentToCommentDto(savedComment);
+    }*/
+/*@Override
+    public CommentDto addComment(CommentDto commentDto) {
+        return commentMapper.commentToCommentDto(commentRepository.save(commentMapper.commentDtoToComment(commentDto)));
+    }
+    @Override
+    public CommentDto addComment(CreateOrUpdateCommentDto commentDto) {
+        return commentMapper.commentToCommentDto(commentRepository.save(commentMapper.createOrUpdateCommentFromDto(commentDto)));
+    }*/
+
+    @Override
+    public void deleteComment( Integer commentId, Integer adId) {
```

Comment:
```suggestion
    public void deleteComment( Integer adId, Integer commentId) {
```
Меняем местами параметры